src/org/cyverse/metadata_files/datacite.clj

```diff
@@ -55,8 +58,9 @@
   (generate [self]
     (if-let [missing (seq (mdf/missing-attributes self))]
       (util/missing-required-attributes missing)
-      (let [values (util/associated-attr-values attributes ["datacite.creator" "creatorAffiliation"])]
-        (Creators. (mapv (fn [[name affiliation]] (Creator. name affiliation)) values))))))
+      (let [values (util/associated-attr-values
+                    attributes ["datacite.creator" "creatorAffiliation"] ["creatorNameIdentifier"])]
+        (Creators. (mapv (fn [[name affiliation name-id]] (Creator. name affiliation name-id)) values))))))
```

Comment:
It kind of works under very specific conditions. This code assumes that the first instance of each attribute contains the information for the first creator, the second contains the information for the second creator, etc. It's not perfect because it'll be really easy to cross the streams if the order of the attributes in the response ever changes for any reason. Since we allow all attributes to be repeated, however, I figured I'd do my best to support multiple grouped attributes, though.

To be honest, I didn't look into the metadata service code to see if it does anything to preserve attribute order. It could very well be that adding multiple values for any set of grouped attributes is a surefire recipe for insanity. 🤣

Reply:
Yeah, it's probably fine for now, and we can support this best-effort which should work most of the time. I think our users and admins are already aware of the current limitations in our metadata and won't usually be adding multiples of these attributes anyway. And if they do, then they might not edit them in a way that mixes up the order (outside the templates).

Reply:
Yeah, I wasn't sure if we could guarantee order or not. I figured that the users had to be aware of our current limitations if adding multiple values for attributes can sometimes cause problems, so adding support for it wouldn't hurt anything. And I wasn't planning to actually tell any users that we supported this, especially with nested attributes on the horizon.

Nested attributes are going to be _really_ nice for stuff like this. I'm anxious to have them available.